File: sn.mini.java.jdbc/src/main/java/sn/mini/java/jdbc/Paging.java

package sn.mini.java.jdbc;


public final class Paging {
    private int page;
    private int rows;
    private int total;
    private int maxPage;

    public int getPage() {
        return Math.max(1, page);
    }

    public void setPage(int page) {
        this.page = page;
    }

    public int getRows() {
        return Math.max(1, rows);
    }

    public void setRows(int rows) {
        this.rows = rows;
    }


    public int getTotal() {
        return Math.max(0, total);
    }

    public void setTotal(int total) {
        this.total = total;
    }

    public int getStart() {
        return (page - 1) * rows;
    }

    public int getMaxPage() {
        return (int) Math.ceil(this.total * 1.0 / this.rows);
    }


    public Paging() {
    }

    public Paging(int page, int rows) {
        this.page = page;
        this.rows = rows;
    }
}

File: sn.mini.kotlin.jdbc/src/main/java/sn/mini/kotlin/jdbc/Paging.kt
package sn.mini.kotlin.jdbc

data class Paging(var page: Int, var rows: Int) {
    init {
        page = if (page < 1) 1 else page
        rows = if (rows < 1) 1 else rows
    }

    var total: Int = 0
        set(_total) {
            this.total = _total
            maxpage = Math.ceil(this.total * 0.1 / this.rows).toInt()
            page = if (maxpage > 0 && page > maxpage) maxpage else page
        }
    var start: Int = 0
        get() {
            return (page - 1) * rows;
        }
        private set

    var maxpage: Int = 0
}
